src/app/FakeLib/NuGet/NugetHelper.fs

```diff
@@ -292,14 +299,20 @@ let private pack parameters nuspecFile =
 let rec private publish parameters = 
     let tracing = enableProcessTracing
     enableProcessTracing <- false
-    let source = 
-        if isNullOrEmpty parameters.PublishUrl then ""
-        else sprintf "-source %s" parameters.PublishUrl
-    
-    let args = sprintf "push \"%s\" %s %s" (parameters.OutputPath @@ packageFileName parameters
-                                            |> FullName) parameters.AccessKey source
-    tracefn "%s %s in WorkingDir: %s Trials left: %d" parameters.ToolPath (replaceAccessKey parameters.AccessKey args) 
-        (FullName parameters.WorkingDir) parameters.PublishTrials
+
+    // Newer NuGet requires source to be always specified, so if PublishUrl is empty,
+    // ignore symbol source - the produced source is broken anyway.
+    let normalize str = if str = "" then null else str
```

Comment:
I thought like `if IsNullOrEmpty str then None else Some str` with appropriate changes in the match. It was just an idea, maybe it just looks more ugly in that case ...